src/modules/launcher/Wox.Infrastructure/Image/WindowsThumbnailProvider.cs

```diff
@@ -97,36 +97,53 @@ public static BitmapSource GetThumbnail(string fileName, int width, int height,
 
         private static IntPtr GetHBitmap(string fileName, int width, int height, ThumbnailOptions options)
         {
-            Guid shellItem2Guid = new Guid(IShellItem2Guid);
-            int retCode = NativeMethods.SHCreateItemFromParsingName(fileName, IntPtr.Zero, ref shellItem2Guid, out IShellItem nativeShellItem);
+            IntPtr hBitmap = IntPtr.Zero;
+            IShellItem nativeShellItem = null;
 
-            if (retCode != 0)
+            try
             {
-                throw Marshal.GetExceptionForHR(retCode);
-            }
+                Guid shellItem2Guid = new Guid(IShellItem2Guid);
+                int retCode = NativeMethods.SHCreateItemFromParsingName(fileName, IntPtr.Zero, ref shellItem2Guid, out nativeShellItem);
 
-            NativeSize nativeSize = new NativeSize
-            {
-                Width = width,
-                Height = height,
-            };
+                if (retCode != 0)
+                {
+                    Log.Error($"Error while creating item. retCode:{retCode} ", MethodBase.GetCurrentMethod().DeclaringType);
+                    throw Marshal.GetExceptionForHR(retCode);
+                }
+
+                NativeSize nativeSize = new NativeSize
+                {
+                    Width = width,
+                    Height = height,
+                };
 
-            HResult hr = ((IShellItemImageFactory)nativeShellItem).GetImage(nativeSize, options, out IntPtr hBitmap);
+                HResult hr = ((IShellItemImageFactory)nativeShellItem).GetImage(nativeSize, options, out hBitmap);
 
             // if extracting image thumbnail and failed, extract shell icon
-            if (options == ThumbnailOptions.ThumbnailOnly && hr == HResult.ExtractionFailed)
-            {
-                hr = ((IShellItemImageFactory)nativeShellItem).GetImage(nativeSize, ThumbnailOptions.IconOnly, out hBitmap);
-            }
+                if (options == ThumbnailOptions.ThumbnailOnly && hr == HResult.ExtractionFailed)
+                {
+                    hr = ((IShellItemImageFactory)nativeShellItem).GetImage(nativeSize, ThumbnailOptions.IconOnly, out hBitmap);
+                }
 
-            Marshal.ReleaseComObject(nativeShellItem);
+                if (hr != HResult.Ok)
+                {
+                    throw Marshal.GetExceptionForHR((int)hr);
+                }
 
-            if (hr == HResult.Ok)
-            {
                 return hBitmap;
             }
-
-            throw new InvalidComObjectException($"Error while extracting thumbnail for {fileName}", Marshal.GetExceptionForHR((int)hr));
+            catch (System.Exception ex)
+            {
+                Log.Exception($"Error while extracting thumbnail for {fileName}", ex, MethodBase.GetCurrentMethod().DeclaringType);
+                throw;
```

Comment:
Can we ever reach finally block if we throw here?

Why throwing without description inside of a catch block?

What icon is used if we throw here?

Reply:
Can we ever reach finally block if we throw here?
-Yes, the finally block will still execute even if there's a throw statement in the catch block. The purpose of the finally block is to guarantee that the code within it gets executed regardless of how the try and catch blocks exit, whether they complete normally or exit with an exception.

So, even if you re-throw an exception within a catch block, the corresponding finally block will run before the exception propagates up the call stack. This ensures that any necessary clean-up can be done.

Why throwing without description inside of a catch block?
-We are re-throwing the exception within the catch block to ensure that it gets propagated up the call stack for further handling by upstream code. The reason we're not adding a description is that the original exception already carries information about what went wrong. By using just throw;, we're preserving the original exception's message and stack trace.

What icon is used if we throw here?
-If the hr value is HResult.ExtractionFailed when trying to extract a thumbnail, it will attempt to extract the shell icon instead. After that if it throws an exception while extracting the shell, icon will be empty